modules/programs/himalaya.nix

```diff
@@ -20,30 +20,49 @@ let
           draft = account.folders.drafts;
           # NOTE: himalaya does not support configuring the name of the trash folder
         };
-
+      } // (lib.optionalAttrs (account.signature.showSignature == "append") {
+        # FIXME: signature cannot be attached
+        signature = account.signature.text;
+        signature-delim = account.signature.delimeter;
+      }) // (if account.himalaya.backend == null then {
+        backend = "none";
+      } else if account.himalaya.backend == "imap" then {
         # FIXME: does not support disabling TLS altogether
         # NOTE: does not accept sequence of strings for password commands
+        backend = account.himalaya.backend;
         imap-login = account.userName;
         imap-passwd-cmd = lib.escapeShellArgs account.passwordCommand;
         imap-host = account.imap.host;
         imap-port = account.imap.port;
         imap-starttls = account.imap.tls.useStartTls;
-
+      } else if account.himalaya.backend == "maildir" then {
+        backend = account.himalaya.backend;
+        maildir-root-dir = account.maildirBasePath;
+      } else
+        throw "Unsupported backend: ${account.himalaya.backend}")
+      // (if account.himalaya.sender == null then {
+        sender = "none";
+      } else if account.himalaya.sender == "smtp" then {
+        sender = account.himalaya.sender;
         smtp-login = account.userName;
         smtp-passwd-cmd = lib.escapeShellArgs account.passwordCommand;
         smtp-host = account.smtp.host;
         smtp-port = account.smtp.port;
         smtp-starttls = account.smtp.tls.useStartTls;
-      } // (lib.optionalAttrs (account.signature.showSignature == "append") {
-        # FIXME: signature cannot be attached
-        signature = account.signature.text;
-      }) // account.himalaya.settings;
+      } else if account.himalaya.sender == "sendmail" then
+        if account.himalaya ? sendmail-cmd then {
```

Comment:
You only check the settings existence but don't add it to the configuration?

Reply:
@ambroisie  I'm not entirely sure how to do this in Nix syntax... it's like an ADT where some setting are dependent on both the `sender` and `backend` properties.

```
type Sender =
  | IMAP { imap-port :: Number, ...etc }
  | Sendmail { sendmail-cmd :: String }
```

Reply:
In this case, the `sendmail-cmd` is not needed or used if the `sender` is `imap`.

Reply:
Yes I understood that part, my confusion was just that `sendmail-cmd` seemed to not be set, but that's only because I forgot that `// account.himalaya.settings` is done at the end.

Reply:
Now that I am looking at this again, I'm thinking that perhaps either the `throw` should be removed (and let `himalaya` itself complain about the missing command) or perhaps it should be an actual assertion, using the module system.

Will let the maintainers weight in about what's usually done in home-manager.

Reply:
I'm okay with any of those options.

Reply:
I did the latter... removed the `-cmd` check